pkg/daemon/ceph/client/crush.go

```diff
@@ -154,13 +154,44 @@ func FormatLocation(location, hostName string) ([]string, error) {
 	// set the host name
 	if !isCrushFieldSet("host", pairs) {
 		// keep the fully qualified host name in the crush map, but replace the dots with dashes to satisfy ceph
-		hostName = strings.Replace(hostName, ".", "-", -1)
+		hostName = NormalizeCrushName(hostName)
 		pairs = append(pairs, formatProperty("host", hostName))
 	}
 
 	return pairs, nil
 }
 
+// NormalizeCrushName replaces . with -
+func NormalizeCrushName(name string) string {
```

Comment:
How about unit tests for these normalize functions?

Reply:
+1

Reply:
done